client/my-sites/media-library/upload-url.jsx

```diff
@@ -49,11 +49,13 @@ export default React.createClass( {
 		}
 
 		MediaActions.clearValidationErrors( this.props.site.ID );
-		MediaActions.add( this.props.site.ID, this.state.value );
+		MediaActions.add( this.props.site.ID, this.state.value )
+			.then( () => {
```

Comment:
The problem is that `onAddMedia` (the following line) immediately interrogates the media store for the image that needs to be inserted as preview. If the image needs to be re-oriented, it needs to wait for re-orientation to complete.